app/src/main/java/com/example/evently/data/AccountDB.java

```diff
@@ -0,0 +1,162 @@
+package com.example.evently.data;
+
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.Set;
+import java.util.HashSet;
+import com.example.evently.data.model.Account;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+public class AccountDB {
+
+    // Reference to the accounts collection
+    private final CollectionReference accountsRef;
+
+    private final Set<Account> accounts = new HashSet<Account>();
```

Comment:
In general, database data in its entirety should not be stored in memory. We will never need all accounts throughout the app, we'll only need to query specific accounts by ID.

Though it's probably fine for evently since there's unlikely to be many users, imagine how much memory this could eat up for apps with thousands of users!

Another problem is that bad actors can now inspect the memory of an app and see other accounts. This is more advanced territory and no one will mark us on it but you know..... good to keep in mind for career.

Reply:
Though `addSnapshotListener` seems like the easy route - it'll hurt in the long term since a live snapshot listener is quite costly and should never be used unless real time updates are required (they aren't for us).